src/main/java/cceclipseplugin/core/PluginManager.java

```diff
@@ -463,4 +473,30 @@ public void setAllSubscribedPrefs(boolean b) {
 			e.printStackTrace();
 		}
 	}
+	
+	/**
+	 * Goes through all of the projects in session storage, determines if the user is currently
+	 * subscribed, and then writes that boolean to the "auto-subscribe" preferences.
+	 */
+	public void writeSubscribedProjects() {
+		System.out.println("Writing subscribed projects to auto-subscribe preferences...");
+		SessionStorage ss = PluginManager.getInstance().getDataManager().getSessionStorage();
+		Set<Long> subscribedIDs = ss.getSubscribedIds();
+		List<Project> projects = ss.getProjects();
+		Preferences pluginPrefs = InstanceScope.INSTANCE.getNode(Activator.PLUGIN_ID);
+		Preferences projectPrefs = pluginPrefs.node(PreferenceConstants.NODE_PROJECTS);
+		
+		for (Project p : projects) {
+			boolean subscribed = subscribedIDs.contains(p.getProjectID());
+			Preferences thisProjectPrefs = projectPrefs.node(p.getProjectID() + "");
+			thisProjectPrefs.putBoolean(PreferenceConstants.VAR_SUBSCRIBED, subscribed);
```

Comment:
Why do you put a false entry into the preferences if we only ever get the true entries or check to see if it exists? Shouldn't we just not store something in the preferences for projects we aren't subscribed to?